drivers: uart: Create ambiq uart driver for apollo510

AlessandroLuo opened the pull request:
This commit creates ambiq uart driver for Apollo510 SoC, not to use the pl011 driver any more.

dcpleung commented:
Could you separate this into different commits? For example:

1. HAL change
2. DTS bindings
3. Driver
4. Board DTS changes.

This would make it easier to review.

AlessandroLuo committed:
hal: uart: Add ambiq uart hal support

This commit adds support for ambiq hal uart.

Signed-off-by: Hao Luo <hluo@ambiq.com>

AlessandroLuo committed:
drivers: uart: create ambiq uart driver for apollo510

This commit creates ambiq uart driver for Apollo510 SoC,
not to use the pl011 driver any more.

Signed-off-by: Hao Luo <hluo@ambiq.com>

AlessandroLuo committed:
dts: uart: create ambiq uart binding file

This commit creates ambiq uart new binding file
and renamed the previous one as ambiq,pl011-uart

Signed-off-by: Hao Luo <hluo@ambiq.com>

AlessandroLuo committed:
dts: ambiq: update ambiq boards dtsi as uart binding changed

This commit updates dtsi for ambiq boards as the uart binding
has been changed.

Signed-off-by: Hao Luo <hluo@ambiq.com>

AlessandroLuo commented:
> Could you separate this into different commits? For example:
> 
> 1. HAL change
> 2. DTS bindings
> 3. Driver
> 4. Board DTS changes.
> 
> This would make it easier to review.

@dcpleung done
